R/TealAppDriver.R

```diff
@@ -407,6 +407,30 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
     #' @return Nothing. Opens the underlying teal app in the browser.
     open_url = function() {
       browseURL(self$get_url())
+    },
+    #' @description
+    #' Waits until a specified input, output, or export value.
+    #' This function serves as a wrapper around the `wait_for_value` method,
+    #' providing a more flexible interface for waiting on different types of values within the active module namespace.
+    #' @param ... Dynamic parameters allowing the specification of either an `input`, `output`, or `export`
+    #' parameter with additional value to passed in `wait_for_value`.
+    wait_for_active_module_value = function(...) {
+      dots <- list(...)
+      param_types <- intersect(names(dots), c("input", "output", "export"))
+      checkmate::assert(
+        checkmate::assert_choice(parm_type, c("input", "output", "export")),
+        checkmate::assert_character(dots[[param_type]], len = 1, min.chars = 1),
+        combine = "and"
+      )
+
+
+      modified_id <- sprintf("%s-%s", self$active_module_ns(), dots[[param_type]])
+      dots <- dots[!names(dots) %in% param_type]
+
+      do.call(
+        what = self$wait_for_value,
+        args = c(list(param_type = modified_id, timeout = private$load_timeout), dots)
```

Comment:
Load timeout is not used in this context

> How long to wait for the app to load, in ms. This includes the time to start R. Defaults to 15s

Our own `private$idle_timeout` may be used. Or something different once I get a PR up.

```suggestion
        args = c(list(param_type = modified_id, timeout = private$idle_timeout), dots)
```